application/event_processor/changes.py

```diff
@@ -115,5 +115,13 @@ def update_changes_with_postcode(changes: dict, dos_service: DoSService, nhs_ent
         valid_dos_postcode = get_valid_dos_postcode(nhs_postcode)
         if valid_dos_postcode is None:
             log_invalid_nhsuk_pharmacy_postcode(nhs_entity, dos_service)
+            try:
+                if changes[ADDRESS_CHANGE_KEY]:
+                    del changes[ADDRESS_CHANGE_KEY]
+                    logger.info("Deleted address change as postcode is invalid")
+            except KeyError:
```

Comment:
can this keyerror ever be thrown as you are checking the key exists before deleting it?

Reply:
Yes here - 
![image](https://user-images.githubusercontent.com/62281988/149949355-be88d50f-d16c-4aae-9ca2-abdfcad04dc5.png)
This occurs if the address isn't being changed then it throws the error